tests/test_client_request.py

```diff
@@ -1230,3 +1230,9 @@ def test_loose_cookies_types(loop: Any) -> None:
 def test_gen_default_accept_encoding(has_brotli: Any, expected: Any) -> None:
     with mock.patch("aiohttp.client_reqrep.HAS_BROTLI", has_brotli):
         assert _gen_default_accept_encoding() == expected
+
+
+async def test_certificate(loop: Any, conn: Any) -> None:
+    req = ClientRequest("get", URL("https://www.python.org"), loop=loop)
+    resp = await req.send(conn)
+    assert resp.certificate is not None
```

Comment:
With a local server, you can check/inspect the exact certificate here.

Reply:
Yes, but why should I? This is a test to check if `self.certificate` is populated, not to check if `.get_extra_info("peercert")` extracts the certificate correctly.

Another useful test might be to connect to an HTTP server, such as http://example.com, but  these are getting rare these days.

Reply:
Checking that is it populated _correctly_ would still be preferable.